### ISSUE TITLE
systembridgecli: TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

### ISSUE THREAD

**Issue by ScottG489** [timmo001/system-bridge#1831]
### Description

Running the example `python -m systembridgecli --help` (though on my system I use `python3`) I get the following error:

```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgecli/__main__.py", line 151, in <module>
    app()
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 232, in get_command
    click_command = get_group(typer_instance)
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 218, in get_group
    group = get_group_from_info(TyperInfo(typer_instance))
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 349, in get_group_from_info
    command = get_command_from_info(command_info=command_info)
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 425, in get_command_from_info
    ) = get_params_convertors_ctx_param_name_from_function(command_info.callback)
  File "/home/scott/.local/lib/python3.9/site-packages/typer/main.py", line 401, in get_params_convertors_ctx_param_name_from_function
    parameters = get_params_from_function(callback)
  File "/home/scott/.local/lib/python3.9/site-packages/typer/utils.py", line 11, in get_params_from_function
    type_hints = get_type_hints(func)
  File "/usr/lib/python3.9/typing.py", line 1449, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "/usr/lib/python3.9/typing.py", line 283, in _eval_type
    return t._evaluate(globalns, localns, recursive_guard)
  File "/usr/lib/python3.9/typing.py", line 539, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

### What Platform / OS are you running?

Linux

### What version are you running?

3.3.0

### Anything in the logs or a references that might be useful?

It's not entirely clear to me if `systembridgebackend` needs to be running for the CLI tool to work. It is running via `python3 -m systembridgebackend --no-gui`, but there are some errors. Here's an example of one:

```
2022-07-06 12:24:38 ERROR (Thread-12) [systembridgeshared.database.Database] Error executing SQL: INSERT INTO disk (key, value, timestamp) VALUES ('io_counters_read_count', '992937', 1657124678.8771517) ON CONFLICT(key) DO UPDATE SET value = '992937', timestamp = 1657124678.8771517 WHERE key = 'io_counters_read_count'
cannot commit - no transaction is active
```

All of them seem to be SQL related errors. I can create a separate issue for these errors if it seems unrelated.

### Additional information

Full version information:

```
$ python3 -m pip show systembridgeshared systembridgebackend systembridgecli systembridgefrontend systembridgegui
Name: systembridgeshared
Version: 3.3.0
Summary: System Bridge Shared
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /home/scott/.local/lib/python3.9/site-packages
Requires: aiohttp, appdirs, colorlog, cryptography, incremental, pandas, pydantic
Required-by:
---
Name: systembridgebackend
Version: 3.3.0
Summary: System Bridge Backend
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /home/scott/.local/lib/python3.9/site-packages
Requires: aiogithubapi, incremental, keyboard, mutagen, plyer, psutil, pyinstaller, sanic, sanic-ext, sanic-scheduler, zeroconf
Required-by:
---
Name: systembridgecli
Version: 3.3.0
Summary: System Bridge CLI
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /home/scott/.local/lib/python3.9/site-packages
Requires: incremental, pyinstaller, shellingham, tabulate, typer
Required-by:
---
Name: systembridgefrontend
Version: 3.3.0
Summary: System Bridge Frontend
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /home/scott/.local/lib/python3.9/site-packages
Requires:
Required-by:
---
Name: systembridgegui
Version: 3.3.0
Summary: System Bridge GUI
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /home/scott/.local/lib/python3.9/site-packages
Requires: incremental, pyinstaller, pyperclip, PySide6, typer
Required-by:
```

System info:

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 21.04
Release:        21.04
Codename:       hirsute

$ uname -a
Linux scott-desktop-linux 5.11.0-49-generic #55-Ubuntu SMP Wed Jan 12 17:36:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by ScottG489** [comment]
I left the server running (`python3 -m systembridgebackend --no-gui`) for about 6 hours and came back to this error reporting every 30 seconds:
```
2022-07-06 18:18:38 INFO (MainThread) [scheduler] RUN TASK "_update_data"
2022-07-06 18:18:38 ERROR (MainThread) [scheduler] [Errno 24] Too many open files
2022-07-06 18:18:38 ERROR (MainThread) [scheduler] Traceback (most recent call last):
  File "/home/scott/.local/lib/python3.9/site-packages/sanic_scheduler/__init__.py", line 108, in run
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/server/__init__.py", line 105, in _update_data
    self._data.request_update_data()
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/data.py", line 65, in request_update_data
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/data.py", line 23, in __init__
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/modules/update.py", line 43, in __init__
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/modules/bridge/update.py", line 56, in __init__
  File "/home/scott/.local/lib/python3.9/site-packages/systembridgebackend/modules/bridge/__init__.py", line 17, in __init__
  File "/home/scott/.local/lib/python3.9/site-packages/zeroconf/_core.py", line 450, in __init__
  File "/home/scott/.local/lib/python3.9/site-packages/zeroconf/_utils/net.py", line 378, in create_sockets
  File "/home/scott/.local/lib/python3.9/site-packages/zeroconf/_utils/net.py", line 230, in new_socket
  File "/usr/lib/python3.9/socket.py", line 232, in __init__
OSError: [Errno 24] Too many open files
```

**Comment by rpavlik** [comment]
I think the second issue (the too many open files) is unrelated here. The unsupported operand is, I think, because they're using a syntax for type annotation that was added in Python 3.10, and you (and I) are using Python 3.9.

**Comment by rpavlik** [comment]
Yeah, all these things like this:

```py
    def pixel_clock(
        self,
        database: Database,
        display_key: str,
    ) -> float | None:
```

should instead be:

```py
    def pixel_clock(
        self,
        database: Database,
        display_key: str,
    ) -> Optional[float]:
```
with `from typing import Optional`

**Comment by cpadil** [comment]
I have the exact same issue with Proxmox... I guess the same solution provided by @rpavlik  applies here

### error

```
# python3 -m systembridgecli api-port
Traceback (most recent call last):

  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,

  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)

  File "/usr/local/lib/python3.9/dist-packages/systembridgecli/__main__.py", line 151, in <module>
    app()

  File "/usr/lib/python3.9/typing.py", line 1449, in get_type_hints
    value = _eval_type(value, globalns, localns)

  File "/usr/lib/python3.9/typing.py", line 283, in _eval_type
    return t._evaluate(globalns, localns, recursive_guard)

  File "/usr/lib/python3.9/typing.py", line 539, in _evaluate
    eval(self.__forward_code__, globalns, localns),

  File "<string>", line 1, in <module>

TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

### Some details

```
# lsb_release -a   
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
```

```
# python3 -m pip show systembridgeshared systembridgebackend systembridgecli systembridgefrontend systembridgegui
Name: systembridgeshared
Version: 3.4.2
Summary: System Bridge Shared
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: pydantic, cryptography, appdirs, incremental, colorlog, aiohttp, pandas
Required-by: 
---
Name: systembridgebackend
Version: 3.4.2
Summary: System Bridge Backend
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: sanic-ext, keyboard, mutagen, plyer, incremental, psutil, pyinstaller, aiogithubapi, sanic, zeroconf, sanic-scheduler
Required-by: 
---
Name: systembridgecli
Version: 3.4.2
Summary: System Bridge CLI
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: typer, tabulate, shellingham, incremental, pyinstaller
Required-by: 
---
Name: systembridgefrontend
Version: 3.4.2
Summary: System Bridge Frontend
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: 
Required-by: 
---
Name: systembridgegui
Version: 3.4.2
Summary: System Bridge GUI
Home-page: https://github.com/timmo001/system-bridge
Author: Aidan Timson (Timmo)
Author-email: contact@timmo.xyz
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: pyperclip, typer, PySide6, incremental, pyinstaller
Required-by:
```

**Comment by rpavlik** [comment]
@cpadil did you try my branch and see if it works?

**Comment by ScottG489** [comment]
Confirmed version `3.4.3` has fixed this specific error for me.
